tables/Tables/BaseTable.cc

```diff
@@ -1019,8 +1019,8 @@ BaseTableIterator* BaseTable::makeIterator
 
 const TableDesc& BaseTable::makeEmptyTableDesc() const
 {
-    if (tdescPtr_p.null()) {
-        const_cast<BaseTable*>(this)->tdescPtr_p = new TableDesc();
+    if (!tdescPtr_p) {
+        const_cast<BaseTable*>(this)->tdescPtr_p.reset (new TableDesc());
```

Comment:
Please consider removing this `const_cast` in another PR, or, if removing is not possible, document why the `const_cast` is needed in that PR.

Reply:
I agree const_cast is not nice and should be removed. However, it requires quite some functions to be made non-const, which is too much for now.
I plan to go through all const_cast in casacore and see if they can be changed.